### PR TITLE
[Fixes #77] Made HttpMethod match case-INsensitive

### LINKED ISSUE #77: Make HTTP method match as case-insensitive
I know the spec says that HTTP methods are case-sensitive - but don't we do this other places?

https://github.com/aspnet/CORS/blob/dev/src/Microsoft.AspNetCore.Cors/Infrastructure/CorsMiddleware.cs#L103

### PR DESCRIPTION
**Pull request by kichalla**

### CODE REVIEW

**Comment by rynowak** [comment]
tests?

**Comment by kichalla** [comment]
:up: :date:

File: src/Microsoft.AspNetCore.Cors/Infrastructure/CorsService.cs [hunk 1]
```diff
@@ -109,7 +109,9 @@ public virtual void EvaluatePreflightRequest(HttpContext context, CorsPolicy pol
             var requestHeaders =
                 context.Request.Headers.GetCommaSeparatedValues(CorsConstants.AccessControlRequestHeaders);
 
-            if (!policy.AllowAnyMethod && !policy.Methods.Contains(accessControlRequestMethod))
+            if (!policy.AllowAnyMethod
+                && !policy.Methods.Any((method) =>
+                string.Equals(method, accessControlRequestMethod, StringComparison.OrdinalIgnoreCase)))
```
Comment (rynowak) [inline src/Microsoft.AspNetCore.Cors/Infrastructure/CorsService.cs]: this is really weird formatting - I'd suggest just not using linq here

File: src/Microsoft.AspNetCore.Cors/Infrastructure/CorsMiddleware.cs [hunk 2]
```diff
@@ -100,7 +100,7 @@ public class CorsMiddleware
                     if (string.Equals(
                             context.Request.Method,
                             CorsConstants.PreflightHttpMethod,
-                            StringComparison.Ordinal) &&
+                            StringComparison.OrdinalIgnoreCase) &&
```
Comment (rynowak) [inline src/Microsoft.AspNetCore.Cors/Infrastructure/CorsMiddleware.cs]: This needs a test also

**Comment by kichalla** [comment]
:up: :date:

**Comment by rynowak** [comment]
:shipit:

**Comment by kichalla** [comment]
15b66cec1fac1b12614e23f556bb01b26eb4476a

**Pull request closed without merging**